ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/HeartbeatAgent.java

```diff
@@ -237,19 +241,25 @@ private void processHeartbeats(final long windowStart, final long windowEnd) {
       if (receivedHeartbeats.isEmpty()) {
         hostsStatus.forEach((host, status) -> {
           if (!host.equals(localHostString)) {
-            status.setHostAlive(false);
+            hostsStatus.put(host, new HostStatusEntity(status.getHostInfoEntity(),
```

Comment:
Given that we are updating only a boolean, which is an atomic operation, I feel like this is an overkill. Moreover, I am changing the map to be an AtomicReference of an ImmutableMap with copyOnWrite to avoid the overhead of copyOnRead since we expect more reads than writes. With the new change, there is no reason to worry about concurrency here, no?